File: session06_GrowArray_LinkedList_Stack_Queue/Stack/Stack.cc
#include <iostream>
using namespace std;

class Stack {
private:
	LinkedList1 impl; // head only, one direction
public:
	void push(int v) { // O(1)
		impl.addStart(v); // O(1)
	}
	int pop() { //O(1)
		return impl.removeStart();
	}
	int top() const { //O(1)
		return impl.start();
	}
	bool isEmpty() const { // O(1)
		return impl.isEmpty();
	}
};

File: session06_GrowArray_LinkedList_Stack_Queue/Stack/Stack.java
public class Stack<T> {
	private LinkedList1<T> impl;

	public Stack() { impl = new LinkedList1<>(); }
	public void push(T val) { //O(1)
    impl.addStart(val); //O(1)
	}
	
	public T pop() {
		return impl.removeStart(); //O(1)
	}
	
	public boolean isEmpty() {
		return impl.isEmpty();
	}
}
